lib/resty/core/ndk.lua

```diff
@@ -0,0 +1,91 @@
+-- Copyright (C) by OpenResty Inc.
+
+
+local ffi = require 'ffi'
+local base = require "resty.core.base"
+
+
+local C = ffi.C
+local ffi_cast = ffi.cast
+local ffi_new = ffi.new
+local ffi_str = ffi.string
+local FFI_OK = base.FFI_OK
+local get_string_buf = base.get_string_buf
+local get_request = base.get_request
+local getmetatable = getmetatable
+local type = type
+local tostring = tostring
+local error = error
+
+
+local _M = {
+    version = base.version
+}
+
+
+ffi.cdef[[
+typedef unsigned char u_char;
```

Comment:
Other modules in this repository make a point not to alias `unsigned char` to `u_char`. E.g. https://github.com/openresty/lua-resty-core/blob/1d5c898e3574d7c38ef2d8d905e8481f9cfd5aaa/lib/resty/core/hash.lua#L16-L17

Not sure if we should start doing so here.